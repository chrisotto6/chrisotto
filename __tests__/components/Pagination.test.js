import { render } from '@testing-library/react'
import Pagination from '@/components/Pagination'

it('renders Pagination unchanged', () => {
  const { container } = render(<Pagination totalPage="4" currentPage="2" />)
  expect(container).toMatchSnapshot()
})

it('renders Pagination with one page unchanged', () => {
  const { container } = render(<Pagination totalPage="1" currentPage="1" />)
  expect(container).toMatchSnapshot()
})

it('renders Pagination on the last page unchanged', () => {
  const { container } = render(<Pagination totalPage="2" currentPage="1" />)
  expect(container).toMatchSnapshot()
})
