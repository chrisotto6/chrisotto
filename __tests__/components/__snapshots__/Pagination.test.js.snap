// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pagination on the last page unchanged 1`] = `
<div>
  <div
    class="space-y-2 pt-6 pb-8 md:space-y-5"
  >
    <nav
      class="flex justify-between"
    >
      <button
        class="cursor-auto disabled:opacity-50"
        disabled=""
        rel="previous"
      >
        Previous
      </button>
      <span>
        1
         of 
      </span>
      <button
        class="cursor-auto disabled:opacity-50"
        disabled=""
        rel="next"
      >
        Next
      </button>
    </nav>
  </div>
</div>
`;

exports[`renders Pagination unchanged 1`] = `
<div>
  <div
    class="space-y-2 pt-6 pb-8 md:space-y-5"
  >
    <nav
      class="flex justify-between"
    >
      <a
        href="/blog"
      >
        <button
          rel="previous"
        >
          Previous
        </button>
      </a>
      <span>
        2
         of 
      </span>
      <button
        class="cursor-auto disabled:opacity-50"
        disabled=""
        rel="next"
      >
        Next
      </button>
    </nav>
  </div>
</div>
`;

exports[`renders Pagination with one page unchanged 1`] = `
<div>
  <div
    class="space-y-2 pt-6 pb-8 md:space-y-5"
  >
    <nav
      class="flex justify-between"
    >
      <button
        class="cursor-auto disabled:opacity-50"
        disabled=""
        rel="previous"
      >
        Previous
      </button>
      <span>
        1
         of 
      </span>
      <button
        class="cursor-auto disabled:opacity-50"
        disabled=""
        rel="next"
      >
        Next
      </button>
    </nav>
  </div>
</div>
`;
